pytest_bdd/cucumber_json.py

```diff
@@ -142,7 +186,7 @@ def stepmap(step):
             "description": "",
             "tags": self._serialize_tags(scenario),
             "type": "scenario",
```

Comment:
Hm, I did not add support for scenario outlines, that was already here. What I changed was filling in variables in step names in cases of scenario outlines.

I'm not sure why scenario_outline is not handled, but I'm guessing that's because the "expanded" reporting (that this module was already using) does not output any "scenario_outline", instead it "expands" them into several "scenario".

IMHO I don't think anything has to be changed here. What do you think?

Reply:
I think its a bug in the report. It was written in one day many years ago by just 2 guys. Now that we noticed that we could return "scenario_outline" instead of "scenario" once the scenario has examples.

Reply:
You say we should change the output to the format specified in https://relishapp.com/cucumber/cucumber/docs/formatters/json-output-formatter#scenario-outline instead of https://relishapp.com/cucumber/cucumber/docs/formatters/json-output-formatter#scenario-outline-expanded ?

This is not backwards compatible with the current output:
1) it uses "scenario_outline" instead of "scenario"
2) it includes "examples" (that's an addition, so it might not be a big problem)
3) it does not include any results, only outline + examples, so you can't use this to create a report.

Maybe I'm missing something here?

Reply:
I need a closer look, but it seems that we have a separate ScenarioReport per pytest item. So the examples utilize the pytest paramerization and the items are generated. However each item has its own ScenarioReport with examples of only one record in the table. It would be a challenge to combine them back to implement the "scenario_outline". OK, let's stick with the expanded one as the default format (this way "scenario", not "scenario_outline").

Let's just cleanup and remove the StepName then since you have enough context in the serialization loop.

I'm a bit worried though that we don't implement "scenario_outline". People might debug tests by looking up the step name as it is defined (with <param>), not by values. So the change might be surprising. I'll think of a way to have the "scenario_outline" version of it...

Reply:
Hm, you have a point there (looking up tests by non-filled-in step name). How about we append something then in the the report?

Something like:
```
type <type> and value <value>, using type="int" and value="42"
```

In the mean time, I'll take a look into replacing the StepName solution with something else.

Could you give a little bit more explanation about how there is enough context in the serialization loop? I've run the debugger on that point and checked local/self variables and couldn't find it (maybe I missed it).

The only point I could find was in the name of the test, which is generated by pytest, but I hate to rely on that since it's an implementation detail and can change any time (and besides, we have to do some ugly parsing of the test name).